toolchain/check/eval.cpp

```diff
@@ -200,7 +204,13 @@ static auto GetPhase(SemIR::ConstantId constant_id) -> Phase {
     return Phase::Template;
   } else {
     CARBON_CHECK(constant_id.is_symbolic());
-    return Phase::Symbolic;
+    if (eval_context.constant_values()
+            .GetSymbolicConstant(constant_id)
+            .period_self) {
+      return Phase::PeriodSelfSymbolic;
+    } else {
+      return Phase::Symbolic;
+    }
```

Comment:
Maybe make use of the new function here too?
```suggestion
  } else if (eval_context.constant_values().DependsOnGenericParameter(constant_id)) {
    return Phase::Symbolic;
  } else {
    CARBON_CHECK(constant_id.is_symbolic());
    return Phase::PeriodSelfSymbolic;
  }
```
(I'm not sure, your approach might be more readable.)

Reply:
Am I wrong to worry about it doing extra work (calling `constant_id.is_symbolic()` directly and in the call to `DependsOnGenericParameter`)? I guess that the optimizer can see the contents of `DependsOnGenericParameter`, so my concerns are probably unfounded.

Reply:
That was my thinking -- assuming a reasonable amount of inlining happens, the optimizer should be able to combine the two calls to `is_symbolic`. And I *think* it should be a very cheap check anyway, given that it's just testing the sign bit and performing a perfectly-predicted branch on it.